From 2cd62f487e918fad3fd6f9df564b1a0dc843a4df Mon Sep 17 00:00:00 2001
From: Garrett Brown <themagnificentmrb@gmail.com>
Date: Wed, 6 Jun 2018 17:37:44 -0700
Subject: [PATCH] Games: Fix broken savestate location for standalone game
 clients

Log output after applying fix:

   DEBUG: RetroPlayer[PLAYER]: Closing file
   ERROR: Savestates not implemented for standalone game clients
   DEBUG: RetroPlayer[SAVE]: Failed to save state at close
---
 xbmc/games/addons/GameClient.cpp                 | 2 +-
 xbmc/games/addons/savestates/SavestateWriter.cpp | 9 +++++++--
 2 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/xbmc/games/addons/GameClient.cpp b/xbmc/games/addons/GameClient.cpp
index b090fc59958e..f3c90159a681 100644
--- a/xbmc/games/addons/GameClient.cpp
+++ b/xbmc/games/addons/GameClient.cpp
@@ -308,7 +308,7 @@ bool CGameClient::OpenStandalone(RETRO::IStreamManager& streamManager, IGameInpu
     return false;
   }
 
-  if (!InitializeGameplay(ID(), streamManager, input))
+  if (!InitializeGameplay("", streamManager, input))
     return false;
 
   return true;
diff --git a/xbmc/games/addons/savestates/SavestateWriter.cpp b/xbmc/games/addons/savestates/SavestateWriter.cpp
index 34326228a880..cc9da8a348f4 100644
--- a/xbmc/games/addons/savestates/SavestateWriter.cpp
+++ b/xbmc/games/addons/savestates/SavestateWriter.cpp
@@ -42,6 +42,13 @@ CSavestateWriter::~CSavestateWriter() = default;
 
 bool CSavestateWriter::Initialize(const CGameClient* gameClient, uint64_t frameHistoryCount)
 {
+  //! @todo Handle savestates for standalone game clients
+  if (gameClient->GetGamePath().empty())
+  {
+    CLog::Log(LOGERROR, "Savestates not implemented for standalone game clients");
+    return false;
+  }
+
   m_savestate.Reset();
   m_fps = 0.0;
 
@@ -59,8 +66,6 @@ bool CSavestateWriter::Initialize(const CGameClient* gameClient, uint64_t frameH
   m_savestate.SetPlaytimeWallClock(frameHistoryCount / m_fps); //! @todo Accumulate playtime instead of deriving it
 
   m_savestate.SetPath(CSavestateUtils::MakePath(m_savestate));
-  if (m_savestate.Path().empty())
-    CLog::Log(LOGDEBUG, "Failed to calculate savestate path");
 
   if (m_fps == 0.0)
     return false; // Sanity check
